clang/lib/Lex/HeaderMap.cpp

```diff
@@ -157,6 +157,9 @@ std::optional<StringRef> HeaderMapImpl::getString(unsigned StrTabIdx) const {
   unsigned MaxLen = FileBuffer->getBufferSize() - StrTabIdx;
   unsigned Len = strnlen(Data, MaxLen);
 
+  if (MaxLen == 0)
+    return std::nullopt;
```

Comment:
I would add the check immediately after `MaxLen` is assigned. It is a little confusing for the check to follow a use of the value, particularly a use that serves no purpose in this case.